File: 2_keyboard_calculator/src_python_version/calculator.py
import time
import sys
import os
from helper import *


# ------------------------------- MAIN FUNCTION ------------------------------ #
def main ():
    keys_tracker = Keys_Tracker()
    print(f'Started, keypresses will be tracked every {interval} seconds for {session} seconds.')
    print('')
    print('------------INTERVAL STATS----------')
    while (time.time() - keys_tracker.start_time) < keys_tracker.session:
        print(f'Keypresses in {interval} seconds: ', keys_tracker.count_keypresses(keypresses))
        print(f'Top 3 keypresses: ', keys_tracker.top_n_keypresses(keypresses, n))
        print(f'Current session speed:', keys_tracker.calculate_average_speed(keypress_speed), 'characters per minute')
        keys_tracker.store_total_keypresses(keypresses)
        keypresses.clear()
        print('------------------------------------')
    #session info
    print('')
    print('------------SESSION STATS----------')
    print(f'Total keypresses: ', sum(keypresses_total))
    print(f'Top 3 keypresses: ', keys_tracker.top_n_keypresses(keypress_total_dict, n))
    print(f'Final session speed:', keys_tracker.calculate_average_speed(keypress_speed), 'characters per minute')
    print('')

if __name__ == '__main__':
    main()

File: 2_keyboard_calculator/src_go_version/calculator.go
package main

import (
	"os"
	"time"
)

var keypresses = []string{}
var keypresses_total = []string{}

var n int = 3 // number of top keypresses to print

/* ------------------------------- CALCULATOR ------------------------------- */
func calculator() {
	println("")
	println("Started, keypresses will be tracked in", interval, "seconds intervals of", session, "seconds")
	println("")
	println("------INTERVAL STATS------")
	for i := 0; i < session; i += interval {
		go store_keypress()
		time.Sleep(time.Duration(interval) * time.Second)
		removeSpecialChars(keypresses)
		println("Keypress in", interval, "seconds:", len(keypresses))
		keypresses_total = append(keypresses_total, keypresses...)
		// calculate typing speed of current session
		top_n_keypresses(n, keypresses, "interval")
		typing_speed(keypresses_total)
		//clear keypresses list
		keypresses = []string{}
		println("--------------------------")
	}
	println("")
	println("------TOTAL STATS------")
	println("Keypress in", session, "seconds:", len(keypresses_total))
	// top n keypresses_total of list
	top_n_keypresses(n, keypresses_total, "session")
	// calculate typing speed of current session
	typing_speed(keypresses_total)
	println("--------------------------")
	os.Exit(0)
}
